pkg/nsx/services/subnet/compare.go

```diff
@@ -24,6 +24,7 @@ func (subnet *Subnet) Value() data.DataValue {
 	s := &Subnet{
 		Tags:             subnet.Tags,
 		SubnetDhcpConfig: subnet.SubnetDhcpConfig,
+		AdvancedConfig:   subnet.AdvancedConfig,
```

Comment:
Can we only compare gateway_addresses and dhcp_server_address? We do not fill ConnectivityState/EnableVlanExtension when building the Subnet but NSX has default value for them. If we compare the whole AdvancedConfig with the NSX Subnet, I suspect we may always get the changed to be true.